integ/src/nodegroup_provider.rs

```diff
@@ -0,0 +1,623 @@
+    iam_client
+        .delete_instance_profile()
+        .instance_profile_name(&iam_instance_profile_name.clone())
+        .send()
+        .await
+        .context("Unable to create instance profile")?;
+    iam_client
+        .delete_role()
+        .role_name(&iam_instance_profile_name.clone())
+        .send()
+        .await
+        .context("Unable to delete role.")?;
+
+    Ok(())
+}
+
+// =^..^=   =^..^=   =^..^=   =^..^=  =^..^=  Related sub-functions of sources creation and termination   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+// Find the node ami id to use.
+async fn find_ami_id(
+    ssm_client: &aws_sdk_ssm::Client,
+    arch: &str,
+    br_version: &str,
+    eks_version: &str,
+) -> ProviderResult<String> {
+    let parameter_name = format!(
+        "/aws/service/bottlerocket/aws-k8s-{}/{}/{}/image_id",
+        eks_version, arch, br_version
+    );
+    let ami_id = ssm_client
+        .get_parameter()
+        .name(parameter_name)
+        .send()
+        .await
+        .context("Unable to get ami id")?
+        .parameter
+        .context("Unable to get ami id")?
+        .value
+        .context("ami id is missing")?;
+    Ok(ami_id)
+}
+
+/// Determine the instance type to use. If provided use that one. Otherwise, for `x86_64` use `m5.large`
+/// and for `aarch64` use `m6g.large`
+async fn instance_type(ec2_client: &aws_sdk_ec2::Client, node_ami: &str) -> ProviderResult<String> {
+    let arch = ec2_client
+        .describe_images()
+        .image_ids(node_ami)
+        .send()
+        .await
+        .context("Unable to get ami architecture")?
+        .images
+        .context("Unable to get ami architecture")?
+        .get(0)
+        .context("Unable to get ami architecture")?
+        .architecture
+        .clone()
+        .context("Ami has no architecture")?;
+
+    Ok(match arch {
+        ArchitectureValues::X8664 => "m5.large",
+        ArchitectureValues::Arm64 => "m6g.large",
+        _ => "m6g.large",
+    }
+    .to_string())
+}
+
+fn first_subnet_id(subnet_ids: &[String]) -> ProviderResult<String> {
+    subnet_ids
+        .get(0)
+        .map(|id| id.to_string())
+        .context("There are no private subnet ids")
+}
+
+fn tag_specifications(cluster_name: &str) -> LaunchTemplateTagSpecificationRequest {
+    LaunchTemplateTagSpecificationRequest::builder()
+        .resource_type(ResourceType::Instance)
+        .tags(
+            Tag::builder()
+                .key("Name")
+                .value(format!("{}_node", cluster_name))
+                .build(),
+        )
+        .tags(
+            Tag::builder()
+                .key(format!("kubernetes.io/cluster/{}", cluster_name))
+                .value("owned")
+                .build(),
+        )
+        .tags(
+            Tag::builder()
+                .key(INSTANCE_TAG_NAME)
+                .value(INSTANCE_TAG_VALUE)
+                .build(),
+        )
+        .build()
+}
+
+fn userdata(
```

Comment:
resolve conflict on this part. The code is from [IPV6 PR](https://github.com/bottlerocket-os/bottlerocket-update-operator/commit/e803d0eaa2f4be468415ea047627df33201b832a)